src/components/CloseButton.tsx

```diff
@@ -1,33 +1,94 @@
+import { PropsWithChildren } from 'react'
 import { useSelector } from 'react-redux'
 import { css, cx } from '../../styled-system/css'
 import { upperRightRecipe } from '../../styled-system/recipes'
+import { SystemStyleObject } from '../../styled-system/types'
 import fastClick from '../util/fastClick'
 import haptics from '../util/haptics'
 
-/** A close button with an ✕. */
-const CloseButton = ({ onClose, disableSwipeToDismiss }: { onClose: () => void; disableSwipeToDismiss?: boolean }) => {
-  const fontSize = useSelector(state => state.fontSize)
-  const padding = fontSize / 2 + 2
+type CloseButtonProps = {
+  onClose: () => void
+  disableSwipeToDismiss?: boolean
+  cssRaw?: SystemStyleObject
+  style?: React.CSSProperties
+}
+
+/** Base for a close button. */
+const BaseCloseButton = ({
+  onClose,
+  disableSwipeToDismiss,
+  cssRaw,
+  style,
+  children,
+}: PropsWithChildren<CloseButtonProps>) => {
   return (
     <a
       {...fastClick(onClose, { tapDown: haptics.medium })}
       className={cx(
         upperRightRecipe(),
-        css({
-          fontSize: 'sm',
-          color: 'inherit',
-          right: '0',
-          textDecoration: 'none',
-          top: '0',
-        }),
+        css(
+          {
+            // inherit not yet supported by plugin
```

Comment:
This comment (x3) can be removed since we have [whitelisted](https://github.com/cybersemics/em/blob/f010868894f21341c49bfcbf7e5a84e8b699c3dd/eslint.config.js#L194) inherit now.

Reply:
Removed: 82afe88e8fe1fd23130193724cd09c13ccdb4ee5